instance/tests/js/instance_app_spec.js

```diff
@@ -0,0 +1,121 @@
+// OpenCraft -- tools to aid developing and hosting free software projects
+// Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+(function(){
+"use strict";
+
+// Tests //////////////////////////////////////////////////////////////////////
+
+describe('Instance app', function () {
+    var $controller,
+        httpBackend,
+        indexController,
+        instanceList,
+        OpenCraftAPI,
+        $scope;
+
+    beforeEach(function() {
+        window.swampdragon = {
+            onChannelMessage: jasmine.createSpy(),
+            ready: jasmine.createSpy()
+        };
+        angular.mock.module('restangular');
+        module('InstanceApp');
+    });
+
+    describe('Index controller', function() {
+        beforeEach(inject(function($controller, _$httpBackend_, $rootScope, _OpenCraftAPI_) {
+            $scope = $rootScope.$new();
+            httpBackend = _$httpBackend_;
+            OpenCraftAPI = _OpenCraftAPI_;
+
+            // Models
+            instanceList = jasmine.loadFixture('api/instances_list.json');
+            httpBackend.whenGET('/api/v1/openedxinstance/').respond(instanceList);
+
+            indexController = $controller('Index', {$scope: $scope, OpenCraftAPI: OpenCraftAPI});
+            httpBackend.flush(); // Clear calls from the controller init
+        }));
+
+        afterEach(function () {
+            httpBackend.verifyNoOutstandingExpectation();
+            httpBackend.verifyNoOutstandingRequest();
+        });
+
+        describe('$scope.select', function() {
+            it('select the instance', function() {
+                $scope.select('a', 'b');
+                expect($scope.selected.a).toEqual('b');
+            });
+        });
+
+        describe('$scope.updateInstanceList', function() {
+            it('loads the instance list from the API on init', inject(function ($httpBackend) {
```

Comment:
Do we need to inject `$httpBackend` here?  It's not used in the function body, and it is available via `httpBackend` set in `beforeEach()` anyway.